extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts

```diff
@@ -179,12 +183,15 @@ export class PublishDatabaseDialog {
 	}
 
 	public async publishClick(): Promise<void> {
+		TelemetryReporter.sendActionEvent(TelemetryViews.PublishDialog, 'publishClicked');
```

Comment:
We should be sending an event for this already - the `ModalDialogClosed` action event with a reason of `close` should be what you're looking for.